### PR TITLE
add TiDB Philosophy

### PR DESCRIPTION
**Pull request by ran-huang**
Signed-off-by: Ran <huangran@pingcap.com>

### CODE REVIEW

File: five-principles-that-guide-tidb-and-pingcap.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+### Make it work
+
+Before we started to build TiDB, we did some simple user research. What problems were troubling current database users? What kind of database should we build? The research told us that users wanted a distributed database that scales out automatically to get rid of data sharding. To meet this user requirement, we began two years of development.
+
+#### Build a SQL layer
+
+During the two years, we made many choices about how TiDB would evolve. For example, TiDB must adopt a multi-layered architecture; otherwise, we couldn't control its complexity.
+
+**TiDB has two layers: the SQL layer and the key-value (KV) layer. **So which layer should we start with? It didn't matter which one we chose, but we had to start somewhere. What criteria should we use to make the choice?
```
Comment (CaitinChen) [inline five-principles-that-guide-tidb-and-pingcap.md:31]:
```suggestion
**TiDB has two layers: the SQL layer and the key-value (KV) layer.** So which layer should we start with? It didn't matter which one we chose, but we had to start somewhere. What criteria should we use to make the choice?
```

File: five-principles-that-guide-tidb-and-pingcap.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+In September 2015, we released TiDB to the open source community. It didn't have a storage layer and ran on top of HBase. After the system worked as expected, we contacted the users we had surveyed earlier and asked for their feedback.
+
+#### Provide KV storage
+
+But no one would adopt the prototype. Because TiDB didn't have a KV storage layer, users needed to install Hadoop, Zookeeper, the Hadoop Distributed File System (HDFS), and HBase to make it work. That would be a complete disaster, so potential users turned us down. We immediately started developing the distributed KV layer, which in turn brought about new choices and new considerations.
+
+The first consideration was **correctness**. For critical industries, such as banking, insurance, and securities, data correctness is the most important factor of a database. No one would use a database that processes data incorrectly.
+
+Secondly,** the implementation should be simple**. No user would tolerate a product that's difficult to use. One of our early goals was to make TiDB easier to use than HBase, with a smaller code base. Even today, TiDB's code base is less than one tenth the size of HBase's.
```
Comment (CaitinChen) [inline five-principles-that-guide-tidb-and-pingcap.md:43]:
```suggestion
Secondly, **the implementation should be simple**. No user would tolerate a product that's difficult to use. One of our early goals was to make TiDB easier to use than HBase, with a smaller code base. Even today, TiDB's code base is less than one tenth the size of HBase's.
```

File: five-principles-that-guide-tidb-and-pingcap.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
+---
+title: Five Principles that Guide TiDB and Our Company (Part I)
```
Comment (CaitinChen) [inline five-principles-that-guide-tidb-and-pingcap.md:2]: Our Company -> PingCAP?

File: five-principles-that-guide-tidb-and-pingcap.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+author: ['Max Liu']
+date: 2021-01-21
+summary: PingCAP's CEO talks about the philosophy of TiDB's evolution and how PingCAP built TiDB in the past five years.
+tags: ['Architecture']
+categories: ['Company']
+image: /images/blog/tidb-philosophy-five-principles-that-guide-tidb-and-our-company.jpg
+---
+
+![Five Principles that Guide TiDB and Our Company](media/tidb-philosophy-five-principles-that-guide-tidb-and-our-company.jpg)
```
Comment (CaitinChen) [inline five-principles-that-guide-tidb-and-pingcap.md:11]:
```suggestion
![Five Principles that Guide TiDB and PingCAP](media/tidb-philosophy-five-principles-that-guide-tidb-and-our-company.jpg)
```

Comment (CaitinChen) [reply five-principles-that-guide-tidb-and-pingcap.md:11]: Next time, you could use tidb-philosophy-five-principles-that-guide-tidb-and-pingcap.jpg as the image name.

File: five-principles-that-guide-tidb-and-pingcap.md (hunk 3 above)
Comment (ran-huang) [reply five-principles-that-guide-tidb-and-pingcap.md]: Fixed~

File: five-principles-that-guide-tidb-and-pingcap.md (hunk 4 above)
Comment (ran-huang) [reply five-principles-that-guide-tidb-and-pingcap.md]: Thanks. Fixed~

**Review by CaitinChen — APPROVED** [review]
LGTM

**Pull request merged**